packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/types-section/entity-version-update-moda.tsx

```diff
@@ -0,0 +1,92 @@
+import { faArrowRight } from "@fortawesome/free-solid-svg-icons";
+import { Button, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { Box, Stack, Typography } from "@mui/material";
+
+import { Modal } from "../../../../../../components/modals/modal";
+
+interface EntityVersionUpdateModalProps {
```

Comment:
You're not updating the `EntityVersion` here right? You're updating the entity's _type's_ version

Reply:
(applies to the name of the file too)

Reply:
Good catch, we're going to use this for `entity types` as well as `entity type versions` in near future, so renamed it to `EntityTypeUpdateModal` here https://github.com/hashintel/hash/pull/1774/commits/568780a6ff5c91edc987c91d71f78348d97a51d0

Reply:
Seems better :) But still a bit confusing to me as I'd think you were mutating the entity type itself, rather than the type _of_ the entity, but I'm not really sure how we'd improve upon the naming. It's probably fine because it's in the context of the entity-editor

Reply:
> It's probably fine because it's in the context of the entity-editor


I think the same, since based on designs, this modal well be used for "updating entity type version" and "changing entity type". Maybe in future, we can rename this to sth like `review (cannot find a good word here now) changes modal`